third_party/cpgfunctionEP-0.3.2/CMakeLists.txt

```diff
@@ -0,0 +1,86 @@
+cmake_minimum_required(VERSION 3.6)
+project(cpgfunctionEP)
+
+set(CMAKE_CXX_STANDARD 17)
+if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --std=c++17")
+else()
+    set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+    add_compile_definitions(_USE_MATH_DEFINES)
+endif()
+
+# Everything will need to access this
+include_directories(include)
+# Third party includes
+include_directories(third_party/eigen-3.3.9)  # Eigen for LU decomposition
+include_directories(third_party/qdt-master)
+
+# Create a DLL with an API
+add_library(cpgfunctionEP SHARED
```

Comment:
do we have a need for this to be explicitly SHARED? If the goal is to always link it to energypluslib, then STATIC would make more sense. But I would omit the keyword, rely on the value of `BUILD_SHARED_LIBS` to determine if STATIC or SHARED is picked.

Reply:
Believed to be addresssed in a7f66c005851db3116f729a4ad814d4d86fd0d41